stock_inventory_exclude_sublocation/views/stock_inventory_view.xml

```diff
@@ -0,0 +1,18 @@
+<!-- Copyright 2017 Eficent Business and IT Consulting Services S.L.
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+
+<record id="view_inventory_form" model="ir.ui.view">
+    <field name="name">Inventory form view - cycle count extension </field>
+    <field name="model">stock.inventory</field>
+    <field name="inherit_id" ref="stock.view_inventory_form"/>
+    <field name="arch" type="xml">
+        <field name="filter" position="after">
+            <field name="exclude_sublocation"
+                   attrs="{'invisible': [('filter', '!=', 'none')]}"/>
```

Comment:
The remaining option is to select only one product and the field to select the product is this (https://github.com/odoo/odoo/blob/9.0/addons/stock/stock_view.xml#L108). Since the domain is not affected by the location of the inventory adjustment I think that the "exclude sublocation" option is still not applicable.